### ISSUE TITLE
Error 2: [Code 32] DWI_MISSING_BVEC and Error 2: [Code 33] DWI_MISSING_BVAL

### ISSUE THREAD

**Issue by anshuhim20** [bids-standard/legacy-validator#1845]
Hi, @bids-maintenance 
We have used dcm2bids on our environment (newenv) created on Anaconda. The following is our tree structure

```
(newenv) C:\Users\hp>tree D:\test_29
Folder PATH listing for volume New Volume
Volume serial number is 1626-11F5
D:\TEST_29
├───header_json
└───sub-29
    └───ses-001
        ├───anat
        ├───dwi
        └───func
```

Additionally we have also incorporated the participant.tsv, dataset_description.json and readme.txt files in our test_29 folder.
When we used the bids validator (https://bids-standard.github.io/bids-validator/) it thew up the following error:

```
Error 2: [Code 32] DWI_MISSING_BVEC

sub-29_ses-001_acq-AP_run-01_dwi.nii.gz648.292 KB | application/x-gzip
Location:
test_29/sub-29/ses-001/dwi/sub-29_ses-001_acq-AP_run-01_dwi.nii.gz

Reason:
_dwi scans should have a corresponding .bvec file. It can be included in one of the following locations: /dwi.bvec, /acq-AP_dwi.bvec, /run-01_dwi.bvec, /acq-AP_run-01_dwi.bvec, /sub-29/sub-29_dwi.bvec, /sub-29/sub-29_acq-AP_dwi.bvec, /sub-29/sub-29_run-01_dwi.bvec, /sub-29/sub-29_acq-AP_run-01_dwi.bvec, /sub-29/ses-001/sub-29_ses-001_dwi.bvec, /sub-29/ses-001/sub-29_ses-001_acq-AP_dwi.bvec, /sub-29/ses-001/sub-29_ses-001_run-01_dwi.bvec, /sub-29/ses-001/sub-29_ses-001_acq-AP_run-01_dwi.bvec, /sub-29/ses-001/dwi/sub-29_ses-001_dwi.bvec, /sub-29/ses-001/dwi/sub-29_ses-001_acq-AP_dwi.bvec, /sub-29/ses-001/dwi/sub-29_ses-001_run-01_dwi.bvec, /sub-29/ses-001/dwi/sub-29_ses-001_acq-AP_run-01_dwi.bvec

sub-29_ses-001_acq-PA_run-02_dwi.nii.gz695.801 KB | application/x-gzip
Location:
test_29/sub-29/ses-001/dwi/sub-29_ses-001_acq-PA_run-02_dwi.nii.gz

Reason:
_dwi scans should have a corresponding .bvec file. It can be included in one of the following locations: /dwi.bvec, /acq-PA_dwi.bvec, /run-02_dwi.bvec, /acq-PA_run-02_dwi.bvec, /sub-29/sub-29_dwi.bvec, /sub-29/sub-29_acq-PA_dwi.bvec, /sub-29/sub-29_run-02_dwi.bvec, /sub-29/sub-29_acq-PA_run-02_dwi.bvec, /sub-29/ses-001/sub-29_ses-001_dwi.bvec, /sub-29/ses-001/sub-29_ses-001_acq-PA_dwi.bvec, /sub-29/ses-001/sub-29_ses-001_run-02_dwi.bvec, /sub-29/ses-001/sub-29_ses-001_acq-PA_run-02_dwi.bvec, /sub-29/ses-001/dwi/sub-29_ses-001_dwi.bvec, /sub-29/ses-001/dwi/sub-29_ses-001_acq-PA_dwi.bvec, /sub-29/ses-001/dwi/sub-29_ses-001_run-02_dwi.bvec, /sub-29/ses-001/dwi/sub-29_ses-001_acq-PA_run-02_dwi.bvec


Error 2: [Code 33] DWI_MISSING_BVAL

sub-29_ses-001_acq-AP_run-01_dwi.nii.gz648.292 KB | application/x-gzip
Location:
test_29/sub-29/ses-001/dwi/sub-29_ses-001_acq-AP_run-01_dwi.nii.gz

Reason:
_dwi scans should have a corresponding .bval file. It can be included in one of the following locations: /dwi.bval, /acq-AP_dwi.bval, /run-01_dwi.bval, /acq-AP_run-01_dwi.bval, /sub-29/sub-29_dwi.bval, /sub-29/sub-29_acq-AP_dwi.bval, /sub-29/sub-29_run-01_dwi.bval, /sub-29/sub-29_acq-AP_run-01_dwi.bval, /sub-29/ses-001/sub-29_ses-001_dwi.bval, /sub-29/ses-001/sub-29_ses-001_acq-AP_dwi.bval, /sub-29/ses-001/sub-29_ses-001_run-01_dwi.bval, /sub-29/ses-001/sub-29_ses-001_acq-AP_run-01_dwi.bval, /sub-29/ses-001/dwi/sub-29_ses-001_dwi.bval, /sub-29/ses-001/dwi/sub-29_ses-001_acq-AP_dwi.bval, /sub-29/ses-001/dwi/sub-29_ses-001_run-01_dwi.bval, /sub-29/ses-001/dwi/sub-29_ses-001_acq-AP_run-01_dwi.bval

sub-29_ses-001_acq-PA_run-02_dwi.nii.gz695.801 KB | application/x-gzip
Location:
test_29/sub-29/ses-001/dwi/sub-29_ses-001_acq-PA_run-02_dwi.nii.gz

Reason:
_dwi scans should have a corresponding .bval file. It can be included in one of the following locations: /dwi.bval, /acq-PA_dwi.bval, /run-02_dwi.bval, /acq-PA_run-02_dwi.bval, /sub-29/sub-29_dwi.bval, /sub-29/sub-29_acq-PA_dwi.bval, /sub-29/sub-29_run-02_dwi.bval, /sub-29/sub-29_acq-PA_run-02_dwi.bval, /sub-29/ses-001/sub-29_ses-001_dwi.bval, /sub-29/ses-001/sub-29_ses-001_acq-PA_dwi.bval, /sub-29/ses-001/sub-29_ses-001_run-02_dwi.bval, /sub-29/ses-001/sub-29_ses-001_acq-PA_run-02_dwi.bval, /sub-29/ses-001/dwi/sub-29_ses-001_dwi.bval, /sub-29/ses-001/dwi/sub-29_ses-001_acq-PA_dwi.bval, /sub-29/ses-001/dwi/sub-29_ses-001_run-02_dwi.bval, /sub-29/ses-001/dwi/sub-29_ses-001_acq-PA_run-02_dwi.bval
```

As we are not able to validate our dataset, we are facing issues with quality check. Kindly help us resolve this issue.

Regards,
Himanshu

**Comment by effigies** [comment]
It looks like you're missing bval and bvec files. Hard to tell without a full file listing. This seems like an issue for your convertor, though. Unless you actually have correctly named files, the validator is doing its job.

**Comment by anshuhim20** [comment]
Hi @effigies 
Both the bval and bvec files are present in the subfolder 'dwi' of our subject folder and contains the necessary information as well. You have mentioned "a full file listing", would you like to see the naming of all our files? 
The tree structure of our input directory is given below:

```
D:\TEST_29
└───sub-29
└───ses-001
├───anat
├───dwi
└───func
```

The input directory also contained the `dataset_description.json`, `participants.tsv` and `readme.txt` file.

We have followed the naming conventions as per the bids specification document <https://bids-specification.readthedocs.io/en/stable/04-modality-specific-files/01-magnetic-resonance-imaging-data.html> and the bids validator did not throw up a naming error. The errors that showed is the same as described in our earlier query.
Please advice us about how we can resolve this issue.

Regards,
Himanshu

**Comment by effigies** [comment]
> You have mentioned "a full file listing", would you like to see the naming of all our files?

Yes

**Comment by anshuhim20** [comment]
We have also got in touch independently with @bids-maintenance and Dr Dianne Patterson gave her wonderful insights as given below which helped us resolving the issue:
Why don't you give ezBIDS (https://brainlife.io/ezbids/) a try? It is an online converter and may at least give you some different insights?
ezBIDS gave us a hint that bval and bvec files were being associated with the phase encoding (AP and PA) dwi images along with the original dwi file in our dwi folder . Therefore, we removed the phase encoding files from our dwi sub-folder and put it in fmap folder. 
And bids-validator worked without any issue!

Thanks and regards
Himanshu
